Gems/OpenXRVk/Code/Source/OpenXRVkSession.cpp

```diff
@@ -423,8 +424,20 @@ namespace OpenXRVk
         return static_cast<Input*>(GetInput());
     }
 
-    void Session::UpdateXrSpaceLocations(const OpenXRVk::Device& device, XrTime predictedDisplayTime, AZStd::vector<XrView>& xrViews)
+    void Session::OnBeginFrame([[maybe_unused]] XrTime predictedDisplayTime)
     {
-        GetNativeInput()->UpdateXrSpaceLocations(device, predictedDisplayTime, xrViews);
+        if (IsSessionFocused())
+        {
+            // Syncing actions only works if the session is in focused state
+            m_actionsMgr->SyncActions(predictedDisplayTime);
+        }
+
+        m_referenceSpacesMgr->SyncViews(predictedDisplayTime);
+        
+        //Notify the rest of the engine.
+        const auto& viewPoses = m_referenceSpacesMgr->GetViewPoses();
+        AZ::RPI::XRSpaceNotificationBus::Broadcast(&AZ::RPI::XRSpaceNotifications::OnXRSpaceLocationsChanged,
```

Comment:
If this is called each frame should we be using bus to send notification instead of calling directly through XR api?

Reply:
Not sure exactly what you question is, because that's an ebus notification call already.

At the moment it is a notification EBus that I added to the RPI so the Gems/Camera component could get notified about these poses with the correct predicted display time. But this EBUS will be removed from the RPI and I will add the o3de/Gems/Camera functionality to the OpenXRVk Gem, and that way the RPI becomes cleaner because right now it is a retro-fitted EBUs.